allennlp/training/optimizers.py

```diff
@@ -0,0 +1,38 @@
+from typing import Any, Dict, List, Union
+
+import torch
+
+from ..common.checks import ConfigurationError
+
+optimizers = {  # pylint: disable=invalid-name
+        "adam": torch.optim.Adam,
+        "adagrad": torch.optim.Adagrad,
+        "adadelta": torch.optim.Adadelta,
+        "sgd": torch.optim.SGD,
+        "rmsprop": torch.optim.RMSprop,
+
+}
+
+
+def get_optimizer_from_params(model_parameters: List[torch.Tensor],
+                              params: Union[Dict[str, Any], str]) -> torch.optim.Optimizer:
```

Comment:
I changed this as I thought we might need to pass these optimizer parameters to the `Trainer` constructor as dictionary arguments because in order to instantiate the optimizer, we need the parameters of the model, which I couldn't see how to do in a `from_params` call pleasantly. I changed this back though, as maybe there is a way.